### DISCUSSION TITLE
Leveling tracker characters list limitation and further info

### DISCUSSION THREAD

**Discussion by Nunii** [Lailloken/Exile-UI#142] [Ideas & Feedback]
Hi,

First of all - great tool. One of the bests if not the best.

Second, My list of characters was full and I coulnd't find the new char I've just made.
It took me ~40mins of code inspection and debugging to realise that since I was former 5-way host and that the string the tool is looking for is: "is now level " to realise that I won't see my level 1 char.
Also, It seems that I couldn't find my own characters since none of them has leveled recently - only 5-way participants.

It is indeed seems like an extreme edge case, but maybe it's worth to mention that you can't find level 1 characters.
In addition, maybe add an option to view the n-th latest characters so people like me won't have billion characters list (unless we delete the client log).

Attaching an image to show how it looks like:
![image](https://user-images.githubusercontent.com/33713479/206658660-8f224954-87a7-47b8-ac75-bde0331aacca.png)

Regards,
Nunii

**Comment by Lailloken** [comment]
> First of all - great tool. One of the bests if not the best.

Thank you very much, means a lot.

> It took me ~40mins of code inspection and debugging

I'm sorry you had to go through that, the code is not 'review-friendly'.

I can't remember exactly, but there were issues in the past with only scanning for recent characters, so I had to change it. I'll look into it again, thanks for reporting.

**Reply by Nunii** [reply]
You shouldn't be sorry, I'm always glad to learn new things.

Regarding the characters list - I'm not familliar with ahk but it looks like a python.
So the best straight forward solution could be:
Keep the current solution but add an input: num_characters_to_list (defaulted to list len)
and then return list_of_characters[-num_characters_to_list:]
